app/models/UserAnswers.scala

```diff
@@ -89,6 +94,25 @@ case class UserAnswers(
     }
   }
 
+  def setOriginalSDILReturn(originalSDILReturn: SdilReturn)
+                                 (implicit writes: Writes[CorrectReturnUserAnswersData]): Try[UserAnswers] = {
+
+    val jsPath = JsPath \ "originalSDILReturn"
+
+    val updatedData = data.setObject(jsPath, Json.toJson(originalSDILReturn)) match {
+      case JsSuccess(jsValue, _) =>
+        Success(jsValue)
+      case JsError(errors) =>
+        Failure(JsResultException(errors))
+    }
+
+    updatedData.flatMap {
+      d =>
+        val updatedAnswers = copy(data = d)
```

Comment:
What is `copy(data = d)` doing? Looks a little unusual, and it's not obvious to me what it should be copying from/to